contracts/ExampleNFT.cdc

```diff
@@ -12,18 +13,44 @@ pub contract ExampleNFT: NonFungibleToken {
     pub event Withdraw(id: UInt64, from: Address?)
     pub event Deposit(id: UInt64, to: Address?)
 
-    pub resource NFT: NonFungibleToken.INFT {
+    pub resource NFT: NonFungibleToken.INFT, Views.Resolver {
         pub let id: UInt64
 
-        pub var metadata: {String: String}
+        pub let name: String
```

Comment:
Would it be good to include some comments about what each field and function does?